### PR TITLE
Add supporting for vxWorks

### PR DESCRIPTION
**Pull request by BaoshanPang**
r? @maya-cm
cc @n-salim 
cc @UmeshKalappa

### CODE REVIEW

File: libc-test/build.rs [hunk 1]
```diff
@@ -1927,6 +1928,157 @@ fn test_emscripten(target: &str) {
...
+               "sys/wait.h",
+               "netinet/tcp.h",
+               "syslog.h",
+               "termios.h",
+               "time.h",
+               "ucontext.h",
+               "unistd.h",
+               "utime.h",
+               "wchar.h",
+               "errno.h",
```
Comment (maya-cm) [inline libc-test/build.rs:1989]: Please attach proof that this list of headers is complete.

File: libc-test/build.rs [hunk 2]
```diff
@@ -1927,6 +1928,157 @@ fn test_emscripten(target: &str) {
...
+        | "RTLD_DEFAULT"
+        | "EIOA"
+        | "FIONBIO"
+        | "FIOWRITE"
+        | "FIONREAD"
+        | "VXSIM_EWOULDBLOCK" => true,
+        _ => false,
+    });
+
+    /* Fix me */
```
Comment (maya-cm) [inline libc-test/build.rs:2035]: 'Fix me' - Is this a result of copy-paste or is there something that needs to be resolved here?

File: libc-test/build.rs [hunk 3]
```diff
@@ -1927,6 +1928,157 @@ fn test_emscripten(target: &str) {
...
+        | "lchown" | "setpgid" | "fstatat" | "setsid" | "tcgetpgrp"
+        | "tcsetpgrp" | "killpg" | "mlock" | "munlock" | "mlockall"
+        | "munlockall" | "mmap" | "munmap" | "if_nametoindex"
+        | "if_indextoname" | "flock" | "res_init" | "mknod" | "chroot"
+        | "getsid" | "tcdrain" | "tcflow" | "tcflush" | "tcgetsid"
+        | "tcsendbreak" | "mkdtemp" | "nice" | "grantpt" | "posix_openpt"
+        | "ptsname" | "unlockpt" | "strcasestr" | "socketpair"
+        | "pwrite64" | "pread64" | "fork" | "_pathIsAbsolute" | "execvp"
+        | "setgroups" | "signal" | "getpagesize" | "dlclose" | "dlsym"
+        | "dlerror" | "dlopen" | "setrlimit" | "getrlimit" => true,
```
Comment (maya-cm) [inline libc-test/build.rs:2075]: Please indicate how this list of functions was identified.

**Review by maya-cm — COMMENTED** [review]
Is there a test log for this?

File: libc-test/build.rs (hunk 1 above)
Comment (BaoshanPang) [reply libc-test/build.rs:1989]: Pass building:
$ cargo test --no-run --manifest-path libc-test/Cargo.toml --ta\
rget x86_64-wrs-vxworks -j1
   Compiling libc v0.2.64 (/folk/prj-rust-dev/bpang/libc_test/dev-libc/libc)
   Compiling libc-test v0.1.0 (/folk/prj-rust-dev/bpang/libc_test/dev-libc/libc/libc-test)
    Finished dev [unoptimized + debuginfo] target(s) in 13.94s

Comment (BaoshanPang) [reply libc-test/build.rs:1989]: And run pass:
[vxWorks *]# ala-diab-pb19l:/folk/prj-rust-dev/bpang/libc_test/dev-libc/libc/target/x86_64-wrs-vxworks/debug/main-b4e94fcc89001aa7.v\
xe
ala-diab-pb19l:/folk/prj-rust-dev/bpang/libc_test/dev-libc/libc/target/x86_64-wrs-vxworks/debug/main-b4e94fcc89001aa7.vxe
Launching process 'ala-diab-pb19l:/folk/prj-rust-dev/bpang/libc_test/dev-libc/libc/target/x86_64-wrs-vxworks/debug/main-b4e94fcc8900\
1aa7.vxe' ...
Process 'ala-diab-pb19l:/folk/prj-rust-dev/bpang/libc_test/dev-libc/libc/target/x86_64-wrs-vxworks/debug/main-b4e94fcc89001aa7.vxe' \
(process Id = 0x400980ac0) launched.
RUNNING ALL TESTS
PASSED 2488 tests
[vxWorks *]# ala-diab-pb19l:/folk/prj-rust-dev/bpang/libc_test/dev-libc/libc/target/x86_64-wrs-vxworks/debug/cmsg-7dfc47db37d3828c.v\
xe
ala-diab-pb19l:/folk/prj-rust-dev/bpang/libc_test/dev-libc/libc/target/x86_64-wrs-vxworks/debug/cmsg-7dfc47db37d3828c.vxe
Launching process 'ala-diab-pb19l:/folk/prj-rust-dev/bpang/libc_test/dev-libc/libc/target/x86_64-wrs-vxworks/debug/cmsg-7dfc47db37d3\
828c.vxe' ...
Process 'ala-diab-pb19l:/folk/prj-rust-dev/bpang/libc_test/dev-libc/libc/target/x86_64-wrs-vxworks/debug/cmsg-7dfc47db37d3828c.vxe' \
(process Id = 0x400c9d8a0) launched.

running 5 tests
test t::test_cmsg_data ... ok
test t::test_cmsg_firsthdr ... ok
test t::test_cmsg_len ... ok
test t::test_cmsg_nxthdr ... ok
test t::test_cmsg_space ... ok

test result: ok. 5 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

File: libc-test/build.rs (hunk 2 above)
Comment (BaoshanPang) [reply libc-test/build.rs:2035]: The code was actually developed by Umesh. I believe "Fix me" means we don't know how to test them yet so we skip them right now. We will remove them from the "skip list" when we know how to fix it.

**Comment by BaoshanPang** [comment]
> Is there a test log for this?

https://github.com/Wind-River/libc/pull/23#discussion_r332759242

File: libc-test/build.rs (hunk 3 above)
Comment (BaoshanPang) [reply libc-test/build.rs:2075]: @UmeshKalappa 
Can you help to comment on this question?

Comment (BaoshanPang) [reply libc-test/build.rs:2075]: This is error I get after removing "sigqueue", I believe others have such similiar issue:

running: "wr-c++" "-O0" "-ffunction-sections" "-fdata-sections" "-fPIC" "-g" "-fno-omit-frame-pointer" "-m64" "-static" "-Wall" "-We\
xtra" "-Wall" "-Wextra" "-Werror" "-Wno-unused-parameter" "-Wno-type-limits" "-Wno-address-of-packed-member" "-Wno-unknown-warning-o\
ption" "-Wno-deprecated-declarations" "-o" "/folk/prj-rust-dev/bpang/libc_test/dev-libc/libc/target/x86_64-wrs-vxworks/debug/build/l\
ibc-test-4e44ace510e99266/out/main.o" "-c" "/folk/prj-rust-dev/bpang/libc_test/dev-libc/libc/target/x86_64-wrs-vxworks/debug/build/l\
ibc-test-4e44ace510e99266/out/main.c"
cargo:warning=/folk/prj-rust-dev/bpang/libc_test/dev-libc/libc/target/x86_64-wrs-vxworks/debug/build/libc-test-4e44ace510e99266/out/\
main.c:11858:57: error: type specifier missing, defaults to 'int' [-Werror,-Wimplicit-int]
cargo:warning=            int (*__test_fn_sigqueue(void))(pid_t, int, sigval) {
cargo:warning=                                                        ^
cargo:warning=/folk/prj-rust-dev/bpang/libc_test/dev-libc/libc/target/x86_64-wrs-vxworks/debug/build/libc-test-4e44ace510e99266/out/\
main.c:11859:24: error: incompatible pointer types returning 'int (pid_t, int, const union sigval)' (aka 'int (int, int, const union\
 sigval)') from a function with result type 'int (*)(pid_t, int, int)' (aka 'int (*)(int, int, int)') [-Werror,-Wincompatible-pointe\
r-types]
cargo:warning=                return sigqueue;
cargo:warning=                       ^~~~~~~~
cargo:warning=2 errors generated.
exit code: 1

Comment (maya-cm) [reply libc-test/build.rs:2075]: Ok. Is this list is captured somewhere in a defect or something so we have a record of which ones were skipped? If we don't have a tracker issue, we won't ever re-visit it.

Comment (BaoshanPang) [reply libc-test/build.rs:2075]: Yes, these skipped cases are tracked with this record:
https://jira.wrs.com/browse/V7LIBC-967

**Review by maya-cm — APPROVED** [review]

File: libc-test/build.rs (hunk 2 above)
Comment (UmeshKalappa) [reply libc-test/build.rs:2035]: like @BaoshanPang mentioned, we  will refactor it going forward.

**Pull request merged**